src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlVirtualDocumentFactory.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [Shared]
+    [Export(typeof(VirtualDocumentFactory))]
+    internal class HtmlVirtualDocumentFactory : VirtualDocumentFactory
+    {
+        // Internal for testing
+        internal const string HtmlLSPContentTypeName = "htmly";
```

Comment:
I don't think this is the content type you want, this is the one we're using to light up our stuff inside our LSP.  I think you should have your own content type (that we know nothing about) since this is your internal buffer, or use `htmlyLSP` if you're trying to engage the ILanguageClient.

Reply:
Ya I'm trying to engage the LSP client here. So is `htmlyLSP` a content type you guys define or one I'd define and make inherit from `htmly`?

Reply:
We define it [here](https://devdiv.visualstudio.com/DevDiv/_git/WebTools?path=%2Fsrc%2FLanguages%2Fhtml%2FLanguageServer.InProc%2FHtmlyContentTypeDefinitionImpl.cs&version=GBmaster&_a=contents) and use it for [our ILanguageClient export](https://devdiv.visualstudio.com/DevDiv/_git/WebTools?path=%2Fsrc%2FLanguages%2Fhtml%2FVS%2FLanguageServer%2FHtmlyLanguageClient.cs&version=GBmaster&_a=contents)

Reply:
Awesome! I'll update this accordingly